File: src/main/java/com/IconPippi/simtriggers/util/Logger.java
package com.IconPippi.simtriggers.util;

import java.util.Date;

import com.IconPippi.simtriggers.Main;
import com.IconPippi.simtriggers.gui.console.Console;

/**
 * This class contains some logging utilities
 * @author IconPippi
 *
 */
public class Logger {
	
	@SuppressWarnings("deprecation")
	/*
	 * Get the time string
	 */
	private static String getTime() {
		final Date date = new Date();
		return date.getHours()+":"+date.getMinutes()+":"+date.getSeconds();
	}
	
	/**
	 * Logs a given message into the console (Format: [${time} / LOG] ${message})
	 * @param message to log
	 */
	public static void log(String message) {
		System.out.println("["+getTime()+" / LOG] "+message);
		if (Main.simTriggersGUI != null) Console.println("["+getTime()+" / LOG] "+message);
	}
	
	/**
	 * Prints a message into the console formatted as an error (Format: [${time} / ERROR] ${message})
	 * @param error message
	 */
	public static void error(String message) {
		System.out.println("["+getTime()+" / ERROR] "+message);
		if (Main.simTriggersGUI != null) Console.println("["+getTime()+" / ERROR] "+message);
	}
	 
	/**
	 * Prints a message into the console as a major information (Format: [${time} / MAJOR] ${message})
	 * @param major message
	 */
	public static void major(String message) {
		System.out.println("["+getTime()+" / MAJOR] "+message);
		if (Main.simTriggersGUI != null) Console.println("["+getTime()+" / MAJOR] "+message);
	}
	
	/**
	 * Prints a message into the console as a debug information (Format: [${time} / DEBUG] ${message})
	 * @param debug message
	 */
	public static void debug(String message) {
		System.out.println("["+getTime()+" / DEBUG] "+message);
		if (Main.simTriggersGUI != null) Console.println("["+getTime()+" / DEBUG] "+message);
	}
}

File: src/main/kotlin/dev/iconpippi/logger/Logger.kt
package dev.iconpippi.logger

import java.util.*

/**
 * 01/08/2019
 * This class contains all the logging methods
 *
 * @author IconPippi
 */
object Logger {

    private const val RED = "\u001B[31m"
    private const val YELLOW = "\u001B[33m"
    private const val BLUE = "\u001B[34m"
    private const val PURPLE = "\u001B[35m"
    private const val RESET = "\u001B[0m"

    /**
     * Print a white log message into the console
     * FORMAT: [$time / LOG] $message
     */
    @JvmStatic fun log(msg: String) {
        System.out.println("[${Date().hours}:${Date().minutes}:${Date().seconds} / LOG] $msg$RESET")
    }

    /**
     * Print a red error message into the console
     * FORMAT: [$time / ERROR] $message
     */
    @JvmStatic fun error(msg: String) {
        System.out.println("$RED[${Date().hours}:${Date().minutes}:${Date().seconds} / ERROR] $msg$RESET")
    }

    /**
     * Print a yellow warning message into the console
     * FORMAT: [$time / WARNING] $message
     */
    @JvmStatic fun warning(msg: String) {
        System.out.println("$YELLOW[${Date().hours}:${Date().minutes}:${Date().seconds} / WARNING] $msg$RESET")
    }

    /**
     * Print a blue major message into the console
     * FORMAT: [$time / MAJOR] $message
     */
    @JvmStatic fun major(msg: String) {
        System.out.println("$BLUE[${Date().hours}:${Date().minutes}:${Date().seconds} / MAJOR] $msg$RESET")
    }

    /**
     * Print a purple debug message into the console
     * FORMAT: [$time / DEBUG] $message
     */
    @JvmStatic fun debug(msg: String) {
        System.out.println("$PURPLE[${Date().hours}:${Date().minutes}:${Date().seconds} / DEBUG] $msg$RESET")
    }

}